Tests/Validator/ValidatorWrapperTest.php

```diff
@@ -0,0 +1,114 @@
+        $wrappedValidator = $this->prophesize($validatorInterface);
+        $wrappedValidator->validate($value, null, false, false)->willReturn($violations);
+        $validator = new ValidatorWrapper($wrappedValidator->reveal());
+
+        $this->assertSame(0, count($validator->validate($value)));
+    }
+
+    /**
+     * @dataProvider validatorProvider
+     */
+    public function testValidateProperty($validatorInterface)
+    {
+        $value = 'string_to_validate';
+        $property = 'property';
+        $violations = new ConstraintViolationList();
+        $wrappedValidator = $this->prophesize($validatorInterface);
+        $wrappedValidator->validateProperty($value, $property, null)->willReturn($violations);
+        $validator = new ValidatorWrapper($wrappedValidator->reveal());
+
+        $this->assertSame(0, count($validator->validateProperty($value, $property)));
+    }
+
+    /**
+     * @dataProvider validatorProvider
+     */
+    public function testValidatePropertyValue($validatorInterface)
+    {
+        $value = 'string_to_validate';
+        $property = 'property';
+        $violations = new ConstraintViolationList();
+        $wrappedValidator = $this->prophesize($validatorInterface);
+        $wrappedValidator->validatePropertyValue($value, $property, $value, null)->willReturn($violations);
+        $validator = new ValidatorWrapper($wrappedValidator->reveal());
+
+        $this->assertSame(0, count($validator->validatePropertyValue($value, $property, $value)));
+    }
+
+    /**
+     * @dataProvider validatorProvider
+     */
+    public function testValidateValue($validatorInterface)
+    {
+        $value = 'string_to_validate';
+        $constraint = $this->prophesize('Symfony\Component\Validator\Constraint')->reveal();
+        $violations = new ConstraintViolationList();
+        $wrappedValidator = $this->prophesize($validatorInterface);
+        $wrappedValidator->validateValue($value, $constraint, $value, null)->willReturn($violations);
+        $validator = new ValidatorWrapper($wrappedValidator->reveal());
+
+        $this->assertSame(0, count($validator->validateValue($value, $constraint, $value)));
+    }
+
+    /**
+     * @dataProvider validatorProvider
+     */
+    public function testGetMetadataFactory($validatorInterface)
+    {
+        $metadataFactory = $this->prophesize('Symfony\Component\Validato\MetadataFactoryInterface')->reveal();
+        $wrappedValidator = $this->prophesize($validatorInterface);
+        $wrappedValidator->getMetadataFactory()->willReturn($metadataFactory);
+        $validator = new ValidatorWrapper($wrappedValidator->reveal());
+
+        $this->assertSame($metadataFactory, $validator->getMetadataFactory());
+    }
+
+    public function validatorProvider()
+    {
+        return array(
+            array('Symfony\Component\Validator\ValidatorInterface'),
+            array('Symfony\Component\Validator\Validator\RecursiveValidator'),
+        );
```

Comment:
Change this to return the new interface when it exists, the old interface otherwise.